src/box/vinyl.c

```diff
@@ -1714,11 +1714,17 @@ vy_unique_key_validate(struct vy_lsm *lsm, const char *key,
 	if (original_part_count != part_count) {
 		diag_set(ClientError, ER_EXACT_MATCH,
 			 original_part_count, part_count);
-		return -1;
+		goto error;
 	}
 	const char *key_end;
-	return key_validate_parts(lsm->cmp_def, key, part_count, false,
-				  &key_end);
+	if (key_validate_parts(lsm->cmp_def, key, part_count, false,
```

Comment:
I think we can use `exact_key_validate` here.

Reply:
Looks like there is case when we need this:

<details>
<summary>test output snip</summary>

```diff
[002] vinyl/write_iterator.test.lua                                   [ fail ]
[002]
[002] Test failed! Result content mismatch:
[002] --- vinyl/write_iterator.result	Tue Jan 23 16:39:24 2024
[002] +++ /tmp/t/rejects/vinyl/write_iterator.reject	Thu Apr 11 17:03:54 2024
[002] @@ -217,9 +217,11 @@
[002]  ...
[002]  index2:delete{'9'}
[002]  ---
[002] +- error: Invalid key part count in an exact match (expected 2, got 1)
[002]  ...
[002]  index2:delete{'99'}
[002]  ---
[002] +- error: Invalid key part count in an exact match (expected 2, got 1)
[002]  ...
[002]  box.snapshot()
```

</details>

Reply:
This is unexpected. I don't understand why this happens. Please investigate.

Reply:
You are right. I used `cmp_def` to get this error which is incorrect.